pkg/odo/cli/component/push.go

```diff
@@ -90,77 +123,169 @@ func (po *PushOptions) Complete(name string, cmd *cobra.Command, args []string)
+		return errors.Wrap(err, "failed to get current application")
+	}
+	if currApp != *(po.localConfig.ComponentSettings.App) {
+		log.Namef("Setting %s as active application", *(po.localConfig.ComponentSettings.App))
+		if err = application.SetCurrent(po.client, *(po.localConfig.ComponentSettings.App)); err != nil {
+			log.Errorf("Failed to set %s as active application", *(po.localConfig.ComponentSettings.App))
+			return errors.Wrapf(err, "failed to set %s application as current", *(po.localConfig.ComponentSettings.App))
+		}
+		log.Successf("Successfully set %s as active application", *(po.localConfig.ComponentSettings.App))
 	}
 
-	return
-}
+	isCmpExists, err := component.Exists(po.client, *(po.localConfig.ComponentSettings.ComponentName), *(po.localConfig.ComponentSettings.App))
+	if err != nil {
+		return errors.Wrapf(err, "failed to check if component %s exists or not", *(po.localConfig.ComponentSettings.ComponentName))
+	}
 
-// Run has the logic to perform the required actions as part of command
-func (po *PushOptions) Run() (err error) {
-	stdout := color.Output
+	if !isCmpExists {
+		log.Namef("Creating %s component with name %s", *(po.localConfig.ComponentSettings.ComponentType), *(po.localConfig.ComponentSettings.ComponentName))
+		// Classic case of component creation
+		if err = component.CreateComponent(po.client, po.localConfig.ComponentSettings, po.componentContext, stdout); err != nil {
+			log.Errorf("Failed to create component with settings %+v", po.localConfig.ComponentSettings)
+			os.Exit(1)
+		}
+		log.Successf("Successfully created component %s", *(po.localConfig.ComponentSettings.ComponentName))
+		// after component is successfully created, set it as active
+		log.Namef("Setting component %s as active", *(po.localConfig.ComponentSettings.ComponentName))
+
+		if err = component.SetCurrent(*(po.localConfig.ComponentSettings.ComponentName), *(po.localConfig.ComponentSettings.App), *(po.localConfig.ComponentSettings.Project)); err != nil {
+			return errors.Wrapf(err, "failed to set %s as current component", *(po.localConfig.ComponentSettings.ComponentName))
+		}
+		log.Successf("Component '%s' is now set as active component", *(po.localConfig.ComponentSettings.ComponentName))
 
-	log.Namef("Pushing changes to component: %v", po.componentName)
+	} else {
+		log.Namef("Applying component settings %+v to component: %v", po.localConfig, *(po.localConfig.ComponentSettings.ComponentName))
+		// Apply config
+		err = component.ApplyConfig(po.client, po.localConfig.ComponentSettings, po.componentContext, stdout)
+		if err != nil {
+			log.Errorf("Failed to update config to component deployed. Error %+v", err)
+			os.Exit(1)
+		}
+		log.Successf("Successfully applied component settings %+v to component: %v", po.localConfig.ComponentSettings, *(po.localConfig.ComponentSettings.ComponentName))
+	}
+
+	log.Namef("Pushing changes to component: %v of type %s", *(po.localConfig.ComponentSettings.ComponentName), po.sourceType)
```

Comment:
All this above code (all 200 lines), should have more comments as well as newlines to separate different sections, this is *way* too clustered. 

We should throw this into a separate function too.. There's way too many lines.

Reply:
Ack will separate the create parts into a different function